source/base/index_set.cc

```diff
@@ -514,10 +514,11 @@ IndexSet::make_trilinos_map (const MPI_Comm &communicator,
   // we risk to call different Epetra_Map on different processors and the code
   // hangs.
 #ifdef DEAL_II_WITH_MPI
-  bool all_contiguous = is_contiguous();
-  MPI_Allreduce(MPI_IN_PLACE, &all_contiguous, 1, MPI_LOGICAL, MPI_LAND,
+  // C bindings of MPI do not have MPI_BOOL. Note, that MPI_LOGICAL is Fortran type.
+  int all_contiguous = is_contiguous();
+  MPI_Allreduce(MPI_IN_PLACE, &all_contiguous, 1, MPI_INT, MPI_LAND,
```

Comment:
Yes, according to https://www.open-mpi.org/doc/v1.8/man3/MPI_Reduce_local.3.php
There is also MPI_BAND for bitwise AND, but your suggestion is easier for sure.